lib/smart_answer/calculators/self_assessment_penalties.rb

```diff
@@ -214,18 +220,11 @@ def calculate_interest_for_date(date)
     end
 
     def daily_rate(date)
-      # Rate decreased to 2.6% on 7 April 2020
-      rate_change_date = Date.new(2020, 4, 7)
-      # Rate increased to 2.75% on 4 january 2022
-      second_rate_change_date = Date.new(2022, 1, 4)
-
-      if date >= second_rate_change_date
-        0.0275 / 365.0
-      elsif date >= rate_change_date
-        0.026 / 365.0
-      else
-        0.0325 / 365.0
+      rate_for_date = INTEREST_RATES.find do |rate|
+        date >= Time.zone.parse(rate[:start_date]) && date <= Time.zone.parse(rate[:end_date])
       end
+
+      rate_for_date ? rate_for_date[:value] / 365.0 : 0.0325 / 365.0
```

Comment:
This _might_ read easier as an if/else rather than a ternary...but usually I would also go for a ternary so I'm not too concerned.
Nice refactor of the rates 👍

Reply:
Good point - the ternary was a bit of a mouthful with the calculations, have used `if/else` and `present?` on `rate_for_date` to be even more explicit.